### PR TITLE
Restore ?worker and ?worker=1 to mean ?worker=dedicated

### PR DESCRIPTION
**Pull request by kainino0x**
The parsing for `?worker` and `?worker=1` changed with the addition of `?worker=shared` but these should keep being treated the way they were before.

Issue: #3345

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Review by greggman — APPROVED** [review]
lgtm

File: src/common/runtime/helper/options.ts [hunk 1]
```diff
@@ -14,27 +17,49 @@ export function optionEnabled(
...
+  } else if (value === 'service') {
+    return 'service';
+  } else if (value === 'shared') {
+    return 'shared';
+  } else if (value === '' || value === '1' || value === 'dedicated') {
+    return 'dedicated';
+  } else {
+    unreachable('invalid worker= option value');
+  }
+}
```
Comment (beaufortfrancois) [inline src/common/runtime/helper/options.ts:47]:
```suggestion
  }
  unreachable('invalid worker= option value');
}
```

File: src/common/runtime/helper/options.ts [hunk 2]
```diff
@@ -63,9 +88,9 @@ export type OptionsInfos<Type> = Record<keyof Type, OptionInfo>;
 export const kCTSOptionsInfo: OptionsInfos<CTSOptions> = {
   worker: {
     description: 'run in a worker',
-    parser: optionString,
+    parser: optionWorkerMode,
     selectValueDescriptions: [
-      { value: '', description: 'no worker' },
+      { value: '0', description: 'no worker' },
```
Comment (beaufortfrancois) [inline src/common/runtime/helper/options.ts:93]: Shall we update [docs/intro/developing.md](https://github.com/gpuweb/cts/blob/9db2fcf3ba89e90ca1a13bed512ba54954c8d96e/docs/intro/developing.md?plain=1#L59) as well?

**Review by beaufortfrancois — APPROVED** [review]
LGTM with nits

**Pull request merged**